### PR TITLE
add rm, install and mkdir variables to makefile

### PR DESCRIPTION
**Pull request by sheharyaar**
When running `make clean` and other rules that use `rm` to remove fils/folder, these give out errors if the reffered file/folder does not exist. Hence this PR adds `-f` to ignore nonexistent files and arguments.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -14,7 +14,7 @@ all: lunatik_sym.h
 
 clean:
 	make -C ${KDIR} M=${PWD} clean
-	rm lunatik_sym.h
+	rm -f lunatik_sym.h
```
Comment (lneto) [inline Makefile:17]: perhaps a better idea is to put this in a var.. e.g.,
```
RM = rm -f
```

and while there do the same for mkdir, and install.. just a suggestion, I can merge as is if you disagree..

Comment (sheharyaar) [reply Makefile:17]: Yes, that is much better

Comment (sheharyaar) [reply Makefile:17]: I will make the changes

**Comment by sheharyaar** [comment]
`install` hade params with values so I omitted it.

File: Makefile [hunk 2]
```diff
@@ -3,6 +3,8 @@ SCRIPTS_INSTALL_PATH = /lib/modules/lua
 LUNATIK_INSTALL_PATH = /usr/local/sbin
 LUA_API = lua/lua.h lua/lauxlib.h lua/lualib.h
 KDIR ?= ${MODULES_INSTALL_PATH}/build
+RM = rm -f
+MKDIR = mkdir -p
```
Comment (lneto) [inline Makefile:7]: Please, split this in another commit or update the commit message to reflect all your changes..

Also, I think you could add the permission to this var as well..

**Comment by lneto** [comment]
> `install` had params with values so I omitted it.

perhaps.. 
```
INSTALL = install -o root -g root
...
${INSTALL} -m 0755 ...
...
${INSTALL} -m 0644 ...
```

what do you think?

**Comment by sheharyaar** [comment]
Yes, this can be done.

**Comment by sheharyaar** [comment]
Done

**Review by lneto — APPROVED** [review]
